roslibrust/src/ros1/publisher.rs

```diff
@@ -26,8 +28,10 @@ impl<T: RosMessageType> Publisher<T> {
         }
     }
 
-    pub async fn publish(&self, data: &T) -> Result<(), Box<dyn std::error::Error>> {
-        let data = serde_rosmsg::to_vec(&data)?;
+    pub async fn publish(&self, data: &T) -> Result<(), Box<dyn std::error::Error + Send + Sync>> {
```

Comment:
If a RosLibRustError is being mapped here, just use that type instead of boxing it into something with dyn trait bounds.

Reply:
Punt to #118